### PR TITLE
Add Valkey Resource

### PR DESCRIPTION
**Pull request by Zombach**
Continuation of [Add Valkey. #3855](https://github.com/dotnet/aspire/pull/3855)
Based on theme [Add Garnet Resource. #3839](https://github.com/dotnet/aspire/pull/3839)

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/4324)

### CODE REVIEW

**Comment by Zombach** [comment]
@davidfowl @mitchdenny Hello. Look?

File: src/Aspire.Hosting.ValKey/ValKeyBuilderExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    ///
+    /// var multiplexer = builder.Services.BuildServiceProvider()
+    ///                                   .GetRequiredService&lt;IConnectionMultiplexer&gt;();
+    /// 
+    /// var db = multiplexer.GetDatabase();
+    /// db.HashSet("key", [new HashEntry("hash", "value")]);
+    /// var value = db.HashGet("key", "hash");
+    /// </code>
+    /// </example>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
```
Comment (mitchdenny) [inline src/Aspire.Hosting.ValKey/ValKeyBuilderExtensions.cs:47]: Move params under summary.

Comment (Zombach) [reply src/Aspire.Hosting.ValKey/ValKeyBuilderExtensions.cs]: Do we need to move returns the same way?

**Comment by Zombach** [comment]
@eerhardt Hello. Will you look?

File: tests/Aspire.Hosting.Tests/Aspire.Hosting.Tests.csproj [hunk 2]
```diff
@@ -56,10 +57,15 @@
...
 
     <Content Include="$(RepoRoot)src\Schema\aspire-8.0.json" Link="Schema\aspire-8.0.json">
       <CopyToOutputDirectory>Always</CopyToOutputDirectory>
     </Content>
   </ItemGroup>
 
+  <ItemGroup>
+    <Folder Include="ValKey\" />
+  </ItemGroup>
+
```
Comment (eerhardt) [inline tests/Aspire.Hosting.Tests/Aspire.Hosting.Tests.csproj:70]:
```suggestion
```

This isn't necessary.

File: src/Aspire.Hosting.ValKey/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+* https://valkey.io
+* https://github.com/valkey-io/valkey/blob/unstable/README.md
+* https://stackexchange.github.io/StackExchange.Redis/Basics
+* https://github.com/dotnet/aspire/tree/main/src/Components/README.md
+
+## Feedback & contributing
+
+https://github.com/dotnet/aspire
+
+_*ValKey MIT License. Copyright (c) Microsoft Corporation.._
```
Comment (eerhardt) [inline src/Aspire.Hosting.ValKey/README.md:41]: This doesn't seem correct.

File: src/Aspire.Hosting.ValKey/Aspire.Hosting.ValKey.csproj [hunk 4]
```diff
@@ -0,0 +1,19 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>$(NetCurrent)</TargetFramework>
+    <IsPackable>true</IsPackable>
+    <PackageTags>aspire hosting valkey</PackageTags>
+    <Description>Valkey® support for .NET Aspire.</Description>
+    <PackageIconFullPath>$(SharedDir)valkey-icon.png</PackageIconFullPath>
```
Comment (eerhardt) [inline src/Aspire.Hosting.ValKey/Aspire.Hosting.ValKey.csproj:8]: Do we have permission to reuse the icon?

Comment (Zombach) [reply src/Aspire.Hosting.ValKey/Aspire.Hosting.ValKey.csproj]: I didn't ask for permission
Remove

File: src/Aspire.Hosting.ValKey/README.md (hunk 3 above)
Comment (Zombach) [reply src/Aspire.Hosting.ValKey/README.md]: Really
Remove

Comment (Zombach) [reply src/Aspire.Hosting.ValKey/README.md]: https://github.com/orgs/valkey-io/discussions/585
I created a discussion with ValKey.
If you don't mind, please help me explain the purpose of use.

File: src/Aspire.Hosting.ValKey/README.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
+# Aspire.Hosting.ValKey library
```
Comment (eerhardt) [inline src/Aspire.Hosting.ValKey/README.md:1]: (this applies everywhere)

It looks like the stylized version of this project is `Valkey` (lower case `k`). https://github.com/valkey-io/valkey#what-is-valkey

**Comment by eerhardt** [comment]
I see there is no `WithRedisCommander`. Does Redis Commander work with Valkey?

**Comment by Zombach** [comment]
> I see there is no `WithRedisCommander`. Does Redis Commander work with Valkey?

Initially, when I did the same for Garnet, WithRedisCommander didn't work.
But now after the updates. I checked, it works on both Garnet and Valkey.

File: src/Aspire.Hosting.ValKey/ValKeyBuilderExtensions.cs (hunk 1 above)
Comment (mitchdenny) [reply src/Aspire.Hosting.ValKey/ValKeyBuilderExtensions.cs]: I tend to put the `<remarks>` and `<example>` after everything (with remarks being before examples). Its pedantic as I don't think it truly matters but I am trying to drive some consistency which has a qualify of its own :)

File: src/Aspire.Hosting.ValKey/README.md (hunk 5 above)
Comment (Zombach) [reply src/Aspire.Hosting.ValKey/README.md]: Corrected

**Comment by Zombach** [comment]
> I see there is no `WithRedisCommander`. Does Redis Commander work with Valkey?
Can I add this a little later?
And separate PR?
I will do a separate PR on RedisCommander for Garnet.
I will do a separate PR on RedisCommander for Valkey.

A little busy right now.

File: src/Aspire.Hosting.ValKey/README.md (hunk 3 above)
Comment (mitchdenny) [reply src/Aspire.Hosting.ValKey/README.md]: @Zombach we wouldn't block this PR on the icon. If we can get approval for the icon later we can update the package (we would just use the default icon until then).

**Review by mitchdenny — APPROVED** [review]

**Comment by Zombach** [comment]
@mitchdenny Thank you. Thanks to you, I will learn how to structure comments

File: src/Aspire.Hosting.ValKey/README.md (hunk 3 above)
Comment (Zombach) [reply src/Aspire.Hosting.ValKey/README.md]: Great. Thank you!

**Review by eerhardt — APPROVED** [review]
LGTM. Thanks for the contribution, @Zombach!

**Pull request merged**